src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java

```diff
@@ -961,6 +962,56 @@ public Response deleteUserAccount(@Context final HttpServletRequest httpServletR
         }
     }
 
+    /**
+     * Merge a source user into a target user. The source is deleted afterwards.
+     *
+     * @param httpServletRequest
+     *            - for checking permissions
+     * @param userIdMergeDTO
+     *            - the source and target ids to merge
+     * @return no content or a segue error response
+     */
+    @POST
+    @Path("/users/merge")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response mergeUserAccounts(@Context final HttpServletRequest httpServletRequest,
+                                      final UserIdMergeDTO userIdMergeDTO) {
+        try {
+            RegisteredUserDTO currentlyLoggedInUser = this.userManager.getCurrentRegisteredUser(httpServletRequest);
+            if (!isUserAnAdmin(userManager, currentlyLoggedInUser)) {
+                return new SegueErrorResponse(Status.FORBIDDEN,
+                        "You must be logged in as an admin to access this function.").toResponse();
+            }
+
+            if (currentlyLoggedInUser.getId().equals(userIdMergeDTO.getSourceId())) {
+                return new SegueErrorResponse(Status.BAD_REQUEST, "You are not allowed to be the merge source.")
+                        .toResponse();
+            }
+
+            RegisteredUserDTO targetUser = this.userManager.getUserDTOById(userIdMergeDTO.getTargetId());
+            RegisteredUserDTO sourceUser = this.userManager.getUserDTOById(userIdMergeDTO.getSourceId());
+
+            this.userManager.mergeUserAccounts(targetUser, sourceUser);
+            getLogManager().logEvent(currentlyLoggedInUser, httpServletRequest, SegueLogType.MERGE_USER,
```

Comment:
Could this be `ADMIN_MERGE_USER` instead? Only up to now, `MERGE_USER` belogs to the target user, and not the admin doing the merging. Would be good to separate out the routine anonymous merging from the dangerous admin merging. You'll have to add it to the SegueLogType enum.